srcs/runners_manager/vm_creation/aws/AwsManager.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import boto3
+
+from runners_manager.monitoring.prometheus import metrics
+from runners_manager.runner.Runner import Runner
+from runners_manager.runner.Runner import VmType
+from runners_manager.vm_creation.CloudManager import CloudManager
+from runners_manager.vm_creation.CloudManager import create_vm_metric
+from runners_manager.vm_creation.CloudManager import delete_vm_metric
+
+from runners_manager.vm_creation.aws.schema import AwsConfig
+from runners_manager.vm_creation.aws.schema import AwsConfigVmType
+
+logger = logging.getLogger("runner_manager")
+
+class AwsManager(CloudManager):
+    CONFIG_SCHEMA = AwsConfig
+    CONFIG_VM_TYPE_SCHEMA = AwsConfigVmType
+
+    def __init__(
+        self,
+        name: str,
+        settings: dict,
+        redhat_username: str,
+        redhat_password: str,
+        ssh_keys: str,
+    ):
+        super(AwsManager, self).__init__(
+            name, settings, redhat_username, redhat_password, ssh_keys
+        )
+        self.ec2 = boto3.client("ec2")
+    
+    def delete_existing_runner(self, runner: Runner):
+        """Delete an old runner instance from AWS if it exists."""
+
+        for reservation in self.ec2.describe_instances()['Reservations']:
+            for instance in reservation['Instances']:
+                for tag in instance.get('Tags', []):
+                    if tag.get('Key') == 'Name' and tag.get('Value') == {runner.name}:
+                        instance_id = instance.get('InstanceId')
+                        return self.ec2.terminate_instances(InstanceIds=[instance_id])
```

Comment:
Reminder to make use of `runner.vm_id`, something like:
```suggestion
        self.ec2.terminate_instances(InstanceIds=[runner.vm_id])
```

No idea if this works, but could be something similar

Reply:
It's set up